### PR TITLE
feat(wallet-ui): use wallet backend adapter

### PR DESCRIPTION
**Pull request by michaelfig**
## Description

Enable migration to a tighter wallet backend API for experimentation before the on-chain wallet is ready.

### Testing Considerations

@samsiegart I couldn't figure out how to get the test failure in `WalletConnection.test.jsx` to work.  Can you help?

### CODE REVIEW

**Comment by michaelfig** [comment]
The failure is at https://github.com/Agoric/agoric-sdk/runs/4892183216?check_suite_focus=true#step:12:76

**Comment by samsiegart** [comment]
> The failure is at https://github.com/Agoric/agoric-sdk/runs/4892183216?check_suite_focus=true#step:12:76

This is tricky, and I'm not sure why jest works this way, but this seems to fix it: https://github.com/facebook/create-react-app/issues/9935#issuecomment-732971503

**Comment by samsiegart** [comment]
This might be a better fix https://github.com/facebook/jest/issues/9131#issuecomment-680860100, not sure.

I've run into similar issues with mocks/spies returning undefined because I didn't set them up right, but never with a mock module.

**Comment by michaelfig** [comment]
> This might be a better fix [facebook/jest#9131 (comment)](https://github.com/facebook/jest/issues/9131#issuecomment-680860100), not sure.

Yeah!  Thanks, that works well.  Just a minor change and the tests pass.

File: packages/wallet/ui/src/views/Dapps.jsx [hunk 1]
```diff
@@ -23,7 +23,8 @@ export const DappsWithoutContext = ({ dapps }) => {
     const handleClick = event => {
       setAnchorEl(event.currentTarget);
     };
-    const handleClose = () => {
+    const handleClose = event => {
+      event.stopPropagation();
```
Comment (samsiegart) [inline packages/wallet/ui/src/views/Dapps.jsx]: Was this causing a bug? Just wondering because it doesn't seem related to this change

**Review by samsiegart — APPROVED** [review]

File: packages/wallet/ui/src/views/Dapps.jsx (hunk 1 above)
Comment (michaelfig) [reply packages/wallet/ui/src/views/Dapps.jsx]: Ah, that was a mistake.  I'll take that out.
